crates/orderbook/src/api/post_order.rs

```diff
@@ -121,10 +122,20 @@ impl IntoWarpReply for ValidationErrorWrapper {
                 ),
                 StatusCode::BAD_REQUEST,
             ),
-            ValidationError::WrongOwner(owner) => with_status(
+            ValidationError::WrongOwner(signature::Recovered { message, signer }) => with_status(
                 error(
                     "WrongOwner",
-                    format!("Address recovered from signature {owner} does not match from address"),
+                    format!(
+                        "recovered signer {signer} from signing hash {message:?} does not match \
+                         from address"
+                    ),
+                ),
+                StatusCode::BAD_REQUEST,
+            ),
+            ValidationError::InvalidEip1271Signature(hash) => with_status(
+                error(
+                    "InvalidEip1271Signature",
+                    format!("signature for hash {hash:?} is not valid"),
```

Comment:
Should this error message maybe convey that the hash was computed from the order data or do we expect people to know this?